actors/builtin/miner/deadlines.go

```diff
@@ -14,40 +13,66 @@ import (
 type DeadlineInfo struct {
 	CurrentEpoch abi.ChainEpoch // Epoch at which this info was calculated.
 	PeriodStart  abi.ChainEpoch // First epoch of the proving period (<= CurrentEpoch).
-	Index        uint64         // Current deadline index, in [0..WPoStProvingPeriodDeadlines).
+	Index        uint64         // Current deadline index, in [0..WPoStProvingPeriodDeadlines), or WPoStProvingPeriodDeadlines if period elapsed.
 	Open         abi.ChainEpoch // First epoch from which a proof may be submitted, inclusive (>= CurrentEpoch).
 	Close        abi.ChainEpoch // First epoch from which a proof may no longer be submitted, exclusive (>= Open).
 	Challenge    abi.ChainEpoch // Epoch at which to sample the chain for challenge (< Open).
 	FaultCutoff  abi.ChainEpoch // First epoch at which a fault declaration is rejected (< Open).
 }
 
+// Whether the proving period has begun.
 func (d *DeadlineInfo) PeriodStarted() bool {
 	return d.CurrentEpoch >= d.PeriodStart
 }
 
+// Whether the proving period has elapsed.
+func (d *DeadlineInfo) PeriodElapsed() bool {
```

Comment:
If this returns true from `ComputeProvingPeriodDeadline(minerStateFor(ts).periodStart, ts.Height())` in PoSt scheduling logic, am I supposed to abort, should the code just abort, or is there anything smarter it can do?

Reply:
I think aborting is pretty much it. You probably just have to wait for cron, and if the cron doesn't happen any proof will be rejected anyway.

Possibly you could compute the next period start that you know cron will set it to in order to eagerly begin, but I don't think that complexity is warranted at the moment.